File: src/main/java/com/max/Hello.java
package com.max;

//import com.max.student.Student;

import com.max.studentKT.StudentKT;

public class Hello {
    public static void main(String[] args) {
        StudentKT stu = new StudentKT("Hank",60,70);

        System.out.println("Hello JAVA world");
       String s = new String("abcdef");
//        String s = null;
        System.out.println(s.charAt(3));
        System.out.println(s.length());
        System.out.println(s.substring(1,4));
        Person p = new Person( "MaxLiou",66.5f,1.7f);
        p.hello();
        p.weight = 66.5f;
        p.height = 1.7f;
        System.out.println(p.bmi());
        int score = 88;
        System.out.println(score > 80 || score < 90);
        char c = 'A';
        System.out.println(c > 'a');
        /*int age = 19;
        Integer age2 =19;
        char c = '我';
        Character c2 = 'A';
        byte b = 120;
        float weight = 66.5f;
        float weight2 = 68.5f;
        float height = 1.7f;
        boolean adult = true;
        boolean enroll = false;
        String name = "Tom";*/ //Reference data type
    }
}

File: src/main/kotlin/com/max/Hello.kt
package com.max

import com.max.studentKT.StudentKT

fun main(args: Array<String>) {

    val stu = StudentKT("Hank",80,80)
    stu.grading()
    var s:String? = "abcdef" //除非string定義有個?號
    s = null //Kotlin內定為不可為null
    println(s!!.length) //需要出現Exception
    println(s?.get(3))
    println(s?.substring(3))
//    println("Hello Kotlin World")
//    Human().hello()
    val h = Human( weight=68.5f, height=1.7f);
    println(h.bmi())
    println(h.name)
    val score = 88
    println(score > 60)
    val c:Char = 'A'
    println(c.toInt()>60)
    /*var age  = 19; //static
    age=21;
    var weight : Float = 66.5f;
    var name : String;
    name = "Hank";*/
}

class Human(var name: String ="L", var weight : Float, var height :Float){
    init{
        println("test " + weight)
        println("test again $weight")
    }
    //constructor(name:String, weight: Float, height: Float):this(weight, height)

    fun bmi():Float{
        val bmi = weight / (height*height)
        return bmi;
    }

    fun hello(){
        println("Hello Kotlin!")
    }
}